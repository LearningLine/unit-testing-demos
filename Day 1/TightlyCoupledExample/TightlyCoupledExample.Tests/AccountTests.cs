using ApprovalTests;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace TightlyCoupledExample.Tests
{
	[TestClass]
	public class AccountTests
	{
		[TestMethod]
		//[ExpectedException(typeof(ArgumentOutOfRangeException))]
		public void Credit_WhenCalled_IncreasesAccountBalance()
		{
			//Arrange
			decimal creditAmount = 100m;
			decimal initialBalance = 50m;
			var account = new Account(initialBalance);

			//Act
			account.Credit(creditAmount);

			//Assert
			var expected = initialBalance + creditAmount;

			Assert.AreEqual(expected,account.Balance);
		}

		[TestMethod]
		public void Credit_WhenCalled_IncreaseBalanceAndSetsTransactionDate()
		{
			//Arrange
			decimal creditAmount = 100m;
			decimal initialBalance = 50m;
			var account = new Account(initialBalance);

			TimeProvider.Current = new MockTimeProvider();
			
			//Act
			account.Credit(creditAmount);

			//Assert
			Approvals.Verify(account);
		}
	}
}
