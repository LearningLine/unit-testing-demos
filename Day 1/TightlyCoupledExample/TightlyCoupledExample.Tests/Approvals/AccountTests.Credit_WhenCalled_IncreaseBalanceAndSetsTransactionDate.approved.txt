Balance: 150, LastTransactionDate: 1/1/2015 12:00:00 AM